BlueprintUICommonControls/Sources/AttributedLabel.swift

```diff
@@ -43,53 +57,502 @@ public struct AttributedLabel: Element, Hashable {
+                    {
+                        links.append(.init(text: url.absoluteString, range: result.range))
+                    }
+
+                default:
+                    break
+                }
+            }
+
+            return links
+        }
+
+        private func accessibilityLinks(for links: [Link], in string: NSAttributedString) -> [LinkAccessibilityElement] {
+            links
+                .sorted(by: { $0.range.location < $1.range.location })
+                .compactMap { link in
+                    guard NSIntersectionRange(string.entireRange, link.range).length > 0 else {
+                        return nil
+                    }
+
+                    return LinkAccessibilityElement(
+                        container: self,
+                        label: string.attributedSubstring(from: link.range).string,
+                        link: link
+                    )
+                }
+        }
+
+        func applyLinkColors(activeLinks: [Link] = []) {
+            let mutableString = NSMutableAttributedString(attributedString: attributedText ?? .init(string: ""))
+
+            for link in links {
+                mutableString.addAttributes(linkAttributes, range: link.range)
+            }
+
+            for link in activeLinks {
+                mutableString.addAttributes(activeLinkAttributes, range: link.range)
+            }
+
+            attributedText = mutableString
+        }
+
+        override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+            guard let first = touches.first else { return }
+            let touchedLinks = links(at: first.location(in: self))
+            trackingLinks = touchedLinks
+            applyLinkColors(activeLinks: touchedLinks)
+        }
+
+        override func touchesMoved(_ touches: Set<UITouch>, with event: UIEvent?) {
+            guard let first = touches.first, let trackingLinks = trackingLinks else { return }
+            let touchedLinks = links(at: first.location(in: self))
+            let activeLinks = Set(touchedLinks).intersection(trackingLinks)
+            applyLinkColors(activeLinks: Array(activeLinks))
+        }
+
+        override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
+            guard let first = touches.first, let trackingLinks = trackingLinks else { return }
+            let touchedLinks = links(at: first.location(in: self))
+            let activeLinks = Set(touchedLinks).intersection(trackingLinks)
+            for link in activeLinks {
+                linkHandler?.onTap(link: link.text)
+            }
+
+            self.trackingLinks = nil
+            applyLinkColors()
+        }
+
+        override func touchesCancelled(_ touches: Set<UITouch>, with event: UIEvent?) {
+            trackingLinks = nil
+            applyLinkColors()
+        }
+    }
+
+}
+
+extension AttributedLabel {
+    struct Link: Equatable, Hashable {
+        var text: String
```

Comment:
Hmmmm, I think my preference would be to only support URLs here, rather than having a mushier API.

If we find that folks need more flexibility, rather than supporting stringly-typed IDs, that indicates to me that we need to add a closure-based API.

Reply:
ok, I'll update it to only use URLs

Reply:
- [ ] Only use urls